plaso/storage/redis/redis_store.py

```diff
@@ -0,0 +1,488 @@
+    return cursor, items
+
+  # pylint: disable=arguments-differ
+  def Open(self, redis_client=None, url=None):
+    """Opens the store.
+
+    Args:
+      redis_client (Optional[Redis]): Redis client to query. If specified, no
+          new client will be created. If no client is specified a new client
+          will be opened connected to the Redis instance specified by 'url'.
+      url (Optional[str]): URL for a Redis database. If not specified, the
+          DEFAULT_REDIS_URL will be used.
+
+    Raises:
+      IOError: if the store is already connected to a Redis instance.
+      OSError: if the store is already connected to a Redis instance.
+    """
+    if not url:
+      url = self.DEFAULT_REDIS_URL
+
+    if self._redis_client:
+      raise IOError('Redis client already connected')
+
+    if redis_client:
+      self._redis_client = redis_client
+    else:
+      self._redis_client = redis.from_url(url=url, socket_timeout=60)
+
+    client_name = self._GenerateRedisKey('')
+    self._SetClientName(self._redis_client, client_name)
+
+    metadata_key = self._GenerateRedisKey('metadata')
+    if not self._redis_client.exists(metadata_key):
+      self._WriteStorageMetadata()
+
+  def Remove(self):
+    """Removes the contents of the store from Redis."""
+    merging_key = '{0:s}-{1:s}'.format(
+        self._session_identifier, self._MERGING_KEY_NAME)
+    self._redis_client.hdel(merging_key, self._task_identifier)
+
+    sorted_event_key = self._GenerateRedisKey(self._EVENT_INDEX_NAME)
+    self._redis_client.delete(sorted_event_key)
+
+    task_completion_key = self._GenerateRedisKey(
+        self._CONTAINER_TYPE_TASK_COMPLETION)
+    self._redis_client.delete(task_completion_key)
+
+    metadata_key = self._GenerateRedisKey('metadata')
+    self._redis_client.delete(metadata_key)
+
+  @classmethod
+  def ScanForProcessedTasks(
+      cls, session_identifier, url=None, redis_client=None):
+    """Scans a Redis database for processed tasks.
+
+    Args:
+      session_identifier (Optional[str]): session identifier, formatted as
+          a UUID.
+      url (Optional[str]): URL for a Redis database. If not specified,
+          REDIS_DEFAULT_URL will be used.
+      redis_client (Optional[Redis]): Redis client to query. If specified, no
+          new client will be created.
+
+    Returns:
+      list[str]: identifiers of processed tasks.
+    """
+    if not url:
+      url = cls.DEFAULT_REDIS_URL
+
+    if not redis_client:
+      redis_client = redis.from_url(url=url, socket_timeout=60)
+      cls._SetClientName(redis_client, 'processed_scan')
+
+    finalization_key = '{0:s}-{1:s}'.format(
+        session_identifier, cls._FINALIZED_KEY_NAME)
+    try:
+      task_identifiers = redis_client.hkeys(finalization_key)
+    except redis.exceptions.TimeoutError:
+      # If there is a timeout fetching identifiers, we assume that there are
+      # no processed tasks.
+      return []
```

Comment:
this return does not match the types returned on line 449

Reply:
Done